Modules/Filtering/LabelMap/include/itkStatisticsLabelMapFilter.h

```diff
@@ -137,9 +137,10 @@ class ITK_TEMPLATE_EXPORT StatisticsLabelMapFilter
   static constexpr unsigned int
   GetDefaultNumberOfBins()
   {
-    return NumericTraits<FeatureImagePixelType>::IsInteger && sizeof(FeatureImagePixelType) <= 2
-             ? 1 << (8 * sizeof(FeatureImagePixelType))
-             : 128;
+    constexpr size_t bitsShift = std::min(8 * sizeof(FeatureImagePixelType), 8 * sizeof(m_NumberOfBins) - 1);
+
+    return std::is_integral<FeatureImagePixelType>::value && sizeof(FeatureImagePixelType) <= 2 ? 1u << bitsShift
+                                                                                                : 128u;
```

Comment:
Since we are checking whether `sizeof(FeatureImagePixelType) <= 2`, this method could be re-written as:
```c++
  GetDefaultNumberOfBins()
  {
    return std::is_integral<FeatureImagePixelType>::value && sizeof(FeatureImagePixelType) <= 2
             ? 1u << 8 * std::min(2, sizeof(FeatureImagePixelType))
             : 128u;
  }
```